sql-plugin/src/main/scala/org/apache/spark/sql/rapids/stringFunctions.scala

```diff
@@ -996,6 +1016,40 @@ object GpuRegExpUtils {
    countGroups(parseAST(pattern))
   }
 
+  def getChoicesFromRegex(regex: RegexAST): Option[Seq[String]] = {
```

Comment:
I think it would be worth adding a unit test for this method. It will likely help reviewers understand how this is used as well.

Reply:
Added a unit test